--- tkinter.py.orig	2023-03-19 13:35:28
+++ tkinter.py	2023-03-19 13:35:42
@@ -5,6 +5,8 @@
 import os
 import sys
 import textwrap
+os.environ['SETUPTOOLS_USE_DISTUTILS'] = 'stdlib'
+from pkg_resources import parse_version
 
 try:
     from cStringIO import StringIO
@@ -22,8 +24,8 @@
 
     tk = _tkinter.create()
     version_string = tk.call("info", "patchlevel")
-    return tuple(int(x) for x in version_string.split("."))
-
+    v = parse_version(version_string)
+    return (v.major, v.minor, v.micro)
 
 def check(cmd, mf):
     m = mf.findNode("_tkinter")
